app/merginapi.cpp

```diff
@@ -878,14 +899,17 @@ void MerginApi::listProjectsReplyFinished()
   }
   else
   {
-    QString message = QStringLiteral( "Network API error: %1(): %2" ).arg( QStringLiteral( "listProjects" ), r->errorString() );
-    emit networkErrorOccurred( r->errorString(), QStringLiteral( "Mergin API error: listProjects" ) );
-    InputUtils::log( r->url().toString(), QStringLiteral( "FAILED - %1" ).arg( r->errorString() ) );
+    QString serverMsg = extractServerErrorMsg( r->readAll() );
+    QString message = QStringLiteral( "Network API error: %1(): %2. %3" ).arg( QStringLiteral( "listProjects" ), r->errorString(), serverMsg );
+    emit networkErrorOccurred( serverMsg, QStringLiteral( "Mergin API error: listProjects" ) );
+    InputUtils::log( r->url().toString(), QStringLiteral( "FAILED - %1" ).arg( message ) );
     mMerginProjects.clear();
 
-    if ( r->errorString() == QLatin1String( "Host requires authentication" ) )
+    int status = r->attribute( QNetworkRequest::HttpStatusCodeAttribute ).toInt();
+    if ( status == 401 || status == 400 )
     {
-      emit authRequested();
+      emit authFailed();
+      emit notify( serverMsg );
```

Comment:
why changing from authRequested to authFailed+notify ? does it fix a a bug?

Reply:
The code is actually obsolate, will be removed.